onnxruntime/core/graph/function.cc

```diff
@@ -133,46 +194,48 @@ FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
     std::vector<onnxruntime::NodeArg*> inputs;
     std::vector<onnxruntime::NodeArg*> outputs;
     for (auto input : node->InputDefs()) {
-      auto& n_input = sub_graph.GetOrCreateNodeArg(input->Name(), input->TypeAsProto());
+      auto& n_input = function_body_graph.GetOrCreateNodeArg(input->Name(), input->TypeAsProto());
       inputs.push_back(&n_input);
     }
     for (auto output : node->OutputDefs()) {
-      auto& n_output = sub_graph.GetOrCreateNodeArg(output->Name(), output->TypeAsProto());
+      auto& n_output = function_body_graph.GetOrCreateNodeArg(output->Name(), output->TypeAsProto());
       outputs.push_back(&n_output);
     }
-    sub_graph.AddNode(node->Name(), node->OpType(), node->Description(), inputs, outputs, &node->GetAttributes(), node->Domain());
+    function_body_graph.AddNode(node->Name(), node->OpType(), node->Description(), inputs, outputs, &node->GetAttributes(), node->Domain());
   }
 
   for (const auto& input : meta_def->inputs) {
     const ONNX_NAMESPACE::TensorProto* initializer = nullptr;
     if (graph.GetInitializedTensor(input, initializer)) {
-      sub_graph.AddInitializedTensor(*initializer);
+      function_body_graph.AddInitializedTensor(*initializer);
     }
   }
 
   //TODO: if we reuse the nodes in parent graph, maybe we don't need to resolve it.
-  auto status = sub_graph.Resolve();
+  auto status = function_body_graph.Resolve();
   ORT_ENFORCE(status.IsOK(), status.ErrorMessage());
 }
 
 FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
                            const onnxruntime::NodeIndex& node_index,
-                           const ONNX_NAMESPACE::FunctionProto* onnx_func_proto)
+                           const ONNX_NAMESPACE::FunctionProto& onnx_func_proto)
     : parent_graph_(&graph) {
+  // make a copy of the FunctionProto
   onnx_func_proto_ = onnx_func_proto;
```

Comment:
Would be good to say why this is important.

Reply:
Per my understanding, all ops with a FunctionBody seem to share the same FunctionBody struct within a model. Hence, mutating one causes issues when there are multiple of the same ops in the graph. I think this is the root cause. Do you agree ?

Reply:
Added.